docs/how-to-guides/upgrade-openebs.md

```diff
@@ -0,0 +1,284 @@
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  #VERIFY that you have provided a unique name for this upgrade job.
+  #The name can be any valid K8s string for name. This example uses
+  #the following convention: cstor-spc-<flattened-from-to-versions>
+  name: cstor-spc-1120220
+
+  #VERIFY the value of namespace is same as the namespace where openebs components
+  # are installed. You can verify using the command:
+  # `kubectl get pods -n <openebs-namespace> -l openebs.io/component-name=maya-apiserver`
+  # The above command should return status of the openebs-apiserver.
+  namespace: openebs
+spec:
+  backoffLimit: 4
+  template:
+    spec:
+      #VERIFY the value of serviceAccountName is pointing to service account
+      # created within openebs namespace. Use the non-default account.
+      # by running `kubectl get sa -n <openebs-namespace>`
+      serviceAccountName: openebs-operator
+      containers:
+      - name:  upgrade
+        args:
+        - "cstor-spc"
+
+        # --from-version is the current version of the pool
+        - "--from-version=1.12.0"
+
+        # --to-version is the version desired upgrade version
+        - "--to-version=2.2.0"
+
+        # Bulk upgrade is supported
+        # To make use of it, please provide the list of SPCs
+        # as mentioned below
+        - "cstor-pool-openebs-replica-3"
+
+        #Following are optional parameters
+        #Log Level
+        - "--v=4"
+        #DO NOT CHANGE BELOW PARAMETERS
+        env:
+        - name: OPENEBS_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        tty: true
+
+        # the image version should be same as the --to-version mentioned above
+        # in the args of the job
+        image: openebs/m-upgrade:2.2.0
+        imagePullPolicy: Always
+      restartPolicy: OnFailure
+---
+```
+
+Apply the job to start the upgrade for cStor pools:
+
+```bash
+kubectl apply upgrade-cstor-pools-1120-220.yaml
+```
+
+Ensure the job runs to completion.
+
+
+#### Upgrade cStor volumes
+
+Get existing `CStorVolumes` by executing:
+
+```bash
+k get cstorvolumes --all-namespaces
+NAMESPACE   NAME                                       STATUS    AGE     CAPACITY
+openebs     pvc-5b7d8ee3-59b8-4a02-bd6d-a4e66fbecf9f   Healthy   2d21h   50Gi
+openebs     pvc-e8f8b268-b81c-4201-841a-283f557c44a7   Healthy   2d21h   50Gi
+```
+
+
+Create a Kubernetes job to upgrade the existing cStor volumes:
+
+```bash
+# upgrade-cstor-vols-1120-220.yaml
+
+#This is an example YAML for upgrading cstor volume.
+#Some of the values below needs to be changed to
+#match your openebs installation. The fields are
+#indicated with VERIFY
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  #VERIFY that you have provided a unique name for this upgrade job.
+  #The name can be any valid K8s string for name. This example uses
+  #the following convention: cstor-vol-<flattened-from-to-versions>
+  name: cstor-vol-1120220
```

Comment:
also here a little kubectl magic with bash should help us reduce manual edits.

Reply:
I've changed the `name` to `generateName`. Since `generateName` cannot work with `kubectl apply`, the resulting command is also changed to `kubectl create`, which is correct IMO as the intent is to create a job.